File: from-2301-to-2400/2379-minimum-recolors-to-get-k-consecutive-black-blocks/solution.go
func minimumRecolors(blocks string, k int) int {
    cWhite := 0
    p0, p1 := 0, 0

    for p1 < k {
        if blocks[p1] == 'W' {
            cWhite += 1
        }
        p1 += 1
    }
    minWhite := cWhite

    for p1 < len(blocks) {
        if blocks[p1] == 'W' {
            cWhite += 1
        }
        p1 += 1
        if blocks[p0] == 'W' {
            cWhite -= 1
        }
        p0 += 1
        if cWhite < minWhite {
            minWhite = cWhite
        }
    }
    return minWhite
}

File: from-2301-to-2400/2379-minimum-recolors-to-get-k-consecutive-black-blocks/solution.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        cur_white = 0
        for i in range(k):
            if blocks[i] == 'W':
                cur_white += 1

        min_white = cur_white
        for i in range(k, len(blocks)):
            if blocks[i] == 'W':
                cur_white += 1
            if blocks[i - k] == 'W':
                cur_white -= 1
            min_white = min(min_white, cur_white)
        
        return min_white